yt_dlp/extractor/tbsjp.py

```diff
@@ -0,0 +1,161 @@
+            'subtitles': sub,
+        }
+
+
+class TBSJPProgramIE(InfoExtractor):
+    _VALID_URL = r'https://cu\.tbs\.co\.jp/program/(?P<id>[\d]+)'
+    _TESTS = [{
+        'url': 'https://cu.tbs.co.jp/program/23601',
+        'playlist_mincount': 4,
+        'info_dict': {
+            'id': '23601',
+            'categories': ['エンタメ', 'ミライカプセル', '会社', '働く', 'バラエティ', '動画'],
+            'description': '幼少期の夢は大人になって、どう成長したのだろうか？\nそしてその夢は今後、どのように広がっていくのか？\nいま話題の会社で働く人の「夢の成長」を描く',
+            'series': 'ミライカプセル　-I have a dream-',
+            'title': 'ミライカプセル　-I have a dream-'
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        page = self._download_webpage(url, video_id)
+        meta = self._search_json('window.app=', page, 'programme info', video_id)
+
+        programme = traverse_obj(meta, ('falcorCache', 'catalog', 'program', video_id, 'false', 'value'))
+
+        return {
+            '_type': 'playlist',
+            'entries': [self.url_result(f'https://cu.tbs.co.jp/episode/{id}')
+                        for id in traverse_obj(programme, ('custom_data', 'seriesList', 'episodeCode', ...))],
+            **traverse_obj(programme, {
+                'categories': 'keywords',
+                'id': ('tv_episode_info', 'show_content_id'),
+                'description': ('custom_data', 'program_description'),
+                'series': ('custom_data', 'program_name'),
+                'title': ('custom_data', 'program_name'),
+            })}
+
+
+class TBSJPPlaylistIE(InfoExtractor):
+    _VALID_URL = r'https://cu\.tbs\.co\.jp/playlist/(?P<id>[\da-f]+)'
+    _TESTS = [{
+        'url': 'https://cu.tbs.co.jp/playlist/184f9970e7ba48e4915f1b252c55015e',
+        'playlist_mincount': 4,
+        'info_dict': {
+            'title': 'まもなく配信終了',
+            'id': '184f9970e7ba48e4915f1b252c55015e',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        page = self._download_webpage(url, video_id)
+        meta = self._search_json('window.app=', page, 'playlist info', video_id)
+
+        playlist = traverse_obj(meta, ('falcorCache', 'playList', video_id))
+
+        entries = []
+        for entry in traverse_obj(playlist, ('catalogs', 'value')):
+            # it's likely possible to get most/all of the metadata from the playlist page json,
+            # but just going to go with the lazy solution for now
+            content_id = entry.get('content_id')
+            content_type = entry.get('content_type')
+            if content_type == 'tv_show':
+                url = f'https://cu.tbs.co.jp/program/{content_id}'
+            elif content_type == 'tv_episode':
+                url = f'https://cu.tbs.co.jp/episode/{content_id}'
+            else:
+                self.report_warning(f'{content_id}: Unexpected content_type: {content_type}. Skipping.')
+
+            if url:
+                entries.append(self.url_result(url))
```

Comment:
`if url:` is problematic because `url` is already the input URL, so if the first entry's  `content_type` is unrecognized, the extractor will yield a recursively nested playlist. And `url` will never be falsy if `content_type` is unrecognized, because it will still have its value from the last iteration.

Instead we can append to `entries` immediately and avoid reused/unbound vars

(and in the traversal we can filter for dicts with a truthy `content_id` value)
```suggestion
        entries = []
        for entry in traverse_obj(playlist, ('catalogs', 'value', lambda _, v: v['content_id'])):
            # it's likely possible to get most/all of the metadata from the playlist page json instead
            content_id = entry['content_id']
            content_type = entry.get('content_type')
            if content_type == 'tv_show':
                entries.append(self.url_result(
                    f'https://cu.tbs.co.jp/program/{content_id}', TBSJPProgramIE, content_id))
            elif content_type == 'tv_episode':
                entries.append(self.url_result(
                    f'https://cu.tbs.co.jp/episode/{content_id}', TBSJPEpisodeIE, content_id))
            else:
                self.report_warning(f'Skipping "{content_id}" with unsupported content_type "{content_type}"')
```